### DISCUSSION TITLE
Old and new schemas conflict for RSE

### DISCUSSION THREAD

**Discussion by Peariforme** [zowe/zowe-explorer-vscode#2824] [Q&A]
I updated zowe explorer from version v2.13.1 to v2.15.2 and now I have this warning at vscode startup :

![image](https://github.com/zowe/zowe-explorer-vscode/assets/47952322/8195af34-3a76-4eb2-9e91-58cf6681d6ba)

I do not use rse profils, I only have zosmf profile in my team configuration file. I don't understand why I get this warning and what I can do to improve the situation. Do you have some ideas ?

vscode : 1.88.0
zowe explorer : v2.15.2

**Comment by traeok** [comment]
Hi @Peariforme,

This warning appears when a Zowe Explorer extender tries to register an unversioned schema for their profile type, but it already exists on the user's machine with different contents. From the user's perspective, this is simply for awareness and will not affect the behavior of the registered extenders. You can continue to use Zowe Explorer, as the schema is still available to the extender, but it's just to inform the user that the schema was not updated on disk. In this case, one of the Zowe Explorer extenders is trying to contribute the `rse` profile type with different schema contents than the one you have on disk.

With the schema management changes introduced in v2.15.0, extenders should add a schema version to their schema when changing properties on the schema. This allows Zowe Client applications to manage the schema by keeping the latest version on disk.

**Comment by FALLAI-Denis** [comment]
Hi, 

Does this have an impact on the management of “team profiles”?

We have placed in each of our Git repositories, therefore in each of the VS Code / Zowe Explorer workspaces and for our part IBM Z Open Editor, a `zowe.config.json` file containing the usable profiles and a `zowe.schema.json` file containing the schemas for these profiles...

Will it be necessary to update these files manually, or can this be done transparently and automatically?

Thanks.

**Reply by traeok** [reply]
Hi @FALLAI-Denis,

This does not impact team profiles as the extender can still access the latest schema and profile type. Users do not have to make any changes to their Zowe schema. Zowe Explorer and its extenders can still access the latest data for each Zowe config.

However, the warning will continue to appear until the extender that provides the `rse` profile type addresses the missing schema version. When a schema is updated, the extender should update its schema version so that Zowe CLI and Zowe Explorer can keep track of the latest schema changes.

**Reply by FALLAI-Denis** [reply]
Thanks for your reply. I think the extender that provide the rse profile is IBM Z Open Editor.

**Comment by Peariforme** [comment]
Hello,

Thanks for the answer. I think the nextstep IS tonopen an issue to IBM.

**Reply by zFernand0** [reply]
@JillieBeanSim, @phaumer, should we move this to the `IBM z Open Editor -about` repository?

**Reply by JillieBeanSim** [reply]
I don't think so if seeing it from other extenders like FTP and the message comes from ZE not the extender

**Comment by adam-wolfe** [comment]
I've also run into this with the FTP extension.

Two things:
- How would a user resolve this?
- Can we give users a way to hide this warning/stop it from being shown each time Zowe Explorer is initialized?

**Reply by adam-wolfe** [reply]
After discussing w/ @traeok, it seems like this might have been a result of using a couple different builds of the PR that added the extension schema handling. After deleting my extenders.json and reinstalling the FTP extension I no longer get the warning.

**Comment by JillieBeanSim** [comment]
Hey all, I think if we are seeing with the FTP extension too we may want to think of a way to disable the pop up since it is still being presented to the user by Zowe Explorer not the extenders. Similar to when we see the option to install recommended extensions after cloning a repo.

**Reply by JillieBeanSim** [reply]
<img width="453" alt="Screenshot 2024-04-10 at 5 29 53 PM" src="https://github.com/zowe/zowe-explorer-vscode/assets/49491949/c075069c-2ff4-4f3a-abfa-b14fb34340bd">

**Reply by traeok** [reply]
I looked into this and unfortunately, this is not available through the standard UI functions.
I made a quick change that will move the warnings into the log file for now: https://github.com/zowe/zowe-explorer-vscode/pull/2860

Once the schema management APIs return an error code, we can safely filter against specific codes and show UI messages when appropriate.
